src/app/clusters/mode-select-server/mode-select-server.cpp

```diff
@@ -174,97 +257,230 @@ void emberAfModeSelectClusterServerInitCallback(EndpointId endpointId)
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleChangeToMode(ctx, commandData); });
+        break;
+    case ModeSelect::Commands::ChangeToModeWithStatus::Id:
+        ChipLogDetail(Zcl, "ModeSelect: Entering handling ChangeToModeWithStatus");
+
+        HandleCommand<Commands::ChangeToModeWithStatus::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleChangeToModeWithStatus(ctx, commandData); });
+    }
 }
 
-/**
- * Callback for Mode Select Cluster Server Pre Attribute Changed
- * Enabled in src/app/zap-templates/templates/app/helper.js
- * @param attributePath Concrete attribute path to be changed
- * @param attributeType Attribute type
- * @param size          Attribute size
- * @param value         Attribute value
- */
-InteractionModel::Status MatterModeSelectClusterServerPreAttributeChangedCallback(const ConcreteAttributePath & attributePath,
-                                                                                  EmberAfAttributeType attributeType, uint16_t size,
-                                                                                  uint8_t * value)
+
+bool Instance::HasFeature(ModeSelectFeature feature) const
 {
-    const EndpointId endpointId = attributePath.mEndpointId;
-    InteractionModel::Status result;
+    bool success;
+    uint32_t featureMap;
+    success = (GetFeature(&featureMap) == EMBER_ZCL_STATUS_SUCCESS);
 
-    switch (attributePath.mAttributeId)
+    return success && ((featureMap & to_underlying(feature)) != 0);
+}
+
+// List the commands supported by this instance.
+CHIP_ERROR Instance::EnumerateAcceptedCommands(const ConcreteClusterPath & cluster,
+                                               CommandHandlerInterface::CommandIdCallback callback, void * context)
+{
+    callback(ModeSelect::Commands::ChangeToMode::Id, context);
+    if (HasFeature(ModeSelect::ModeSelectFeature::kExt))
     {
-    case ModeSelect::Attributes::StartUpMode::Id:
-        result = verifyModeValue(endpointId, *value);
-        break;
-    case ModeSelect::Attributes::OnMode::Id:
-        result = verifyModeValue(endpointId, *value);
-        break;
-    default:
-        result = InteractionModel::Status::Success;
+        callback(ModeSelect::Commands::ChangeToModeWithStatus::Id, context);
+        callback(ModeSelect::Commands::ChangeToModeResponse::Id, context);
     }
 
-    return result;
+    return CHIP_NO_ERROR;
 }
 
-/**
- * Checks the new mode against the endpoint's supported modes.
- * @param endpointId    endpointId of the endpoint
- * @param newMode       value of the new mode
- * @return              Success status if the value is valid; InvalidValue otherwise.
- */
-static InteractionModel::Status verifyModeValue(const EndpointId endpointId, const uint8_t newMode)
+// Implements the read functionality for non-standard attributes.
```

Comment:
non-standard in what sense?

Reply:
In the sense that they are not simple reads and require custom code. An example of this is the supported modes as this is a list of structures.

Reply:
That's not what "non-standard" usually means.  The term you are probably looking for here is either "complex" or "list-typed" or "attributes not stored in the attribute store" or something.

Reply:
I changed it to "complex".